### PR TITLE
SERP Parser Interview Case

### PR DESCRIPTION
**Pull request by omrsfylmz**

### CODE REVIEW

File: src/parser/regex/execRegex.spec.ts [hunk 1]
```diff
@@ -21,4 +21,9 @@ describe('execRegex Tests', () => {
     const result = execRegex(value, { pattern: '\\d+' });
     expect('26').to.deep.equal(result);
   });
+  it('Case 4: Define it as a regex object', () => {
+    const value = 'omer 1907';
```
Comment (aykutkardas) [inline src/parser/regex/execRegex.spec.ts]: Open source project contributions should not contain personal information. For example, more general and common examples can be used here.

`90m2`
`George Orwell - 1984`

File: src/parser/regex/execRegex.ts [hunk 2]
```diff
@@ -7,9 +7,12 @@ function execRegex(value: string, regex): string {
     if (REGEXES[regex]) {
       $regex = REGEXES[regex];
     }
-  } else {
+  } else if(regex.pattern) {
     const { pattern, flags } = regex;
     $regex = new RegExp(pattern, flags);
+
```
Comment (aykutkardas) [inline src/parser/regex/execRegex.ts:13]: In order to maintain the code standard in open source projects, it should be written in accordance with the rules of the project. For this, the project includes `eslint` and `prettier` files. Also, this project has a script called `lint` that will do this for you.

`yarn lint`

File: src/parser/regex/execRegex.ts [hunk 3]
```diff
@@ -7,9 +7,12 @@ function execRegex(value: string, regex): string {
...
+  }else{
```
Comment (aykutkardas) [inline src/parser/regex/execRegex.ts]: Maybe checking whether the incoming regex object contains a method named exec can prevent some problems that may occur.

```js
} else if (regex?.exec) {
```

File: src/parser/regex/execRegex.spec.ts [hunk 4]
```diff
@@ -22,7 +22,7 @@ describe('execRegex Tests', () => {
     expect('26').to.deep.equal(result);
   });
   it('Case 4: Define it as a regex object', () => {
-    const value = 'omer 1907';
+    const value = 'test 27';
     const result = execRegex(value, /\d+/g);
     expect('1907').to.deep.equal(result);
```
Comment (aykutkardas) [inline src/parser/regex/execRegex.spec.ts:27]: This test seems to always fail.

**Comment by omrsfylmz** [comment]
Thank you for this good feedback. I will be more careful in the future :)

**Pull request closed without merging**